packages/x-date-pickers/src/internals/hooks/useField/useFieldV6TextField.ts

```diff
@@ -390,8 +390,8 @@ export const useFieldV6TextField: UseFieldTextField<false> = (params) => {
   });
 
   const placeholder = React.useMemo(() => {
-    if (inPlaceholder) {
-      return inPlaceholder;
+    if (params.forwardedProps.hasOwnProperty('placeholder')) {
```

Comment:
How about simply doing:

```suggestion
    if (inPlaceholder !== undefined) {
```

I think that it would be more consistent with the other props in our codebase.
Usually when you pass `undefined`, we fallback to the internal value rather than applying `undefined` (for example in `props.value` or `props.views`)

Reply:
My thought to decide to use this approach was to be able to differentiate the scenarios:

- user explicitly passes the placeholder value as undefined  
- user doesn't pass any placeholder prop

In both cases `inPlaceholder` will be undefined, but on the second scenario, we don't pass forward the prop, so the `params.forwardedProps.hasOwnProperty('placeholder')` will be false and it will fall on the default placeholder, on the other hand if users explicitly passes the prop as undefined, I assume they wish to have an empty placeholder. WDYT ?

Reply:
I totally understand your approach
IMHO, we should render the default placeholder when the user renders `<DateField placeholder={undefined} />`, because it's closer from the other props / components behavior (`<DateField minDate={undefined} />` will apply the 01-01-1900 as the min date rather than removing the boundary, `<DateField value={undefined} />` will fallback to the internal state rather than enforcing an empty value, etc...).

Reply:
I agree with @flaviendelangle on this one, let's go for simplicity. 👍
In "my book", `undefined` == no prop.

Reply:
Thanks, folks. Done in [a2bc2be](https://github.com/mui/mui-x/pull/13148/commits/a2bc2be21d36206cf62a3d120323515bec895797)